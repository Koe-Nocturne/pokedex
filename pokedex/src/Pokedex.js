import React from 'react';
import  Pokecard from "./Pokecard"


class Pokedex extends React.Component {
  render() {
    console.log("PP", this.props.pokemonArray);
    return (
      <div>
        {this.props.pokemonArray.map(pokemon => <Pokecard pokemon={pokemon} />)}

      </div>

    )
  };
};
export default Pokedex;