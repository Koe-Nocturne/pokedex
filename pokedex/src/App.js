import React from 'react';
import './index.css';
import Pokedex from "./Pokedex/Pokedex";

class App extends React.Component {
  render() {

    return (
      <div>
        <Pokedex />
      </div>

    )
  };
};

export default App;