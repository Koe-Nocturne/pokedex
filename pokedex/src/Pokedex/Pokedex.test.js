import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json"
import Pokedex from "./Pokedex";
import App from "../App.js";
// import { isMainThread } from "worker_threads";
import {configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


it("renders without crashing", function(){
  shallow(<Pokedex />);
  mount(<Pokedex />);
});

it("matches screenshot.", function(){
  let wrapper = shallow(<Pokedex />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});


