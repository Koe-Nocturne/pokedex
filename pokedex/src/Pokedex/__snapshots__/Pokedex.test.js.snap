// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches screenshot. 1`] = `
<div>
  <Pokecard
    pokemon={
      Object {
        "base_experience": 62,
        "id": 4,
        "name": "Charmander",
        "type": "fire",
      }
    }
  />
  <Pokecard
    pokemon={
      Object {
        "base_experience": 63,
        "id": 7,
        "name": "Squirtle",
        "type": "water",
      }
    }
  />
  <Pokecard
    pokemon={
      Object {
        "base_experience": 72,
        "id": 11,
        "name": "Metapod",
        "type": "bug",
      }
    }
  />
  <Pokecard
    pokemon={
      Object {
        "base_experience": 178,
        "id": 12,
        "name": "Butterfree",
        "type": "flying",
      }
    }
  />
  <Pokecard
    pokemon={
      Object {
        "base_experience": 112,
        "id": 25,
        "name": "Pikachu",
        "type": "electric",
      }
    }
  />
  <Pokecard
    pokemon={
      Object {
        "base_experience": 95,
        "id": 39,
        "name": "Jigglypuff",
        "type": "normal",
      }
    }
  />
  <Pokecard
    pokemon={
      Object {
        "base_experience": 225,
        "id": 94,
        "name": "Gengar",
        "type": "poison",
      }
    }
  />
  <Pokecard
    pokemon={
      Object {
        "base_experience": 65,
        "id": 133,
        "name": "Eevee",
        "type": "normal",
      }
    }
  />
</div>
`;
